bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java

```diff
@@ -191,15 +193,17 @@ public String getFieldType() {
      * @since com.adobe.cq.forms.core.components.models.form 0.0.1
      */
     @Override
-    @JsonInclude(JsonInclude.Include.NON_NULL)
     @Nullable
     public Boolean isVisible() {
-        if (getEditMode()) {
-            return true;
-        }
+        return visible == null || visible;
```

Comment:
data-cmp-visible when inspected in page source was having its value as false due to this, https://github.com/adobe/aem-core-forms-components/blob/master/ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/textinput.html#L28. I have fixed other such issues related to this

Reply:
can you pls test this change by marking an accordion as hidden?

Reply:
There's one issue with the css approach:
It's upto the consumer of the json to consume the `visible` property. the customer need not use the `data-cmp-visible` in her customization and can write a simple code like:
```
if (visible === false) {
 this.element.style.display = "none";
}
```
She can even use a different data attribute to set for visible property in her custom code and she might rely on a css for the same.
Now she will have to keep in mind to not include this js/css file in the authoring otherwise, it will be an issue that's hard to debug for her.

Reply:
We've implemented a CSS rule that utilizes the !important declaration, indicating its priority over other rules. Here's the rule:

`
form:not(.cmp-adaptiveform-container--edit) [data-cmp-visible="false"] {
    display: none !important;
}
`

If someone opts not to use the `data-cmp-visible` attribute, the visibility behavior may not function correctly. This is because, at the view layer, we rely on these data attributes to deduce the component's visibility. Currently, the attribute is optional, and the form still operates, but visibility won't function as expected. We've made only a few attributes mandatory, governing the model's initialization.

Reply:
Any reason for customising by setting `style.display` instead of `data-cmp-visible` that is documented. I don't see a strong reason, but if they are then they can handle the authoring case as well.